src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelStateDictionary.cs

```diff
@@ -714,6 +709,10 @@ public static bool StartsWithPrefix(string prefix, string key)
                     return false;
                 }
             }
+            else if (key.Length == prefix.Length)
+            {
+                return true;
```

Comment:
`key.StartsWith(prefix, StringComparison.OrdinalIgnoreCase)` has returned `true` so if their lengths are the same they are equal?